internal/engine/errors/errors.go

```diff
@@ -107,6 +107,35 @@ type ActionsError struct {
 	AlertMeta     json.RawMessage
 }
 
+// AsRemediationStatus returs the remediation status from the actions error
+func (ae ActionsError) AsRemediationStatus() db.RemediationStatusTypes {
+	if ae.RemediateErr != nil {
+		return ErrorAsRemediationStatus(ae.RemediateErr)
+	}
+
+	// If there is no metadata, we return success as there are no errors
+	if ae.RemediateMeta == nil {
+		return db.RemediationStatusTypesSuccess
+	}
+
+	remMetaData := map[string]any{}
```

Comment:
You could also declare a struct type here to document what you're going to unpack.

```golang
ret := struct{Status string}{}
```

Doing this would avoid needing to check as much of the contents and types; you could just say `if ret.Status != ""`.